### PR TITLE
Add Default Message Mapper

### PR DESCRIPTION
**Pull request by preardon**
First look into Default Message Mapper

#3355

### CODE REVIEW

File: src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionMessageMapperRegistry.cs [hunk 1]
```diff
@@ -28,27 +28,23 @@ THE SOFTWARE. */
...
-    public class ServiceCollectionMessageMapperRegistry
+    public class ServiceCollectionMessageMapperRegistry(
+        IServiceCollection serviceCollection,
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
     {
-        private readonly IServiceCollection _serviceCollection;
-        private readonly ServiceLifetime _lifetime;
-
         public Dictionary<Type, Type> Mappers { get; } = new Dictionary<Type, Type>();
+        public Type DefaultMessageMapper { get; private set; } = typeof(JsonMessageMapper<>);
```
Comment (iancooper) [inline src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionMessageMapperRegistry.cs:44]: Do we need to default the default here, or do we just do that in the extensions DSLs?

File: src/Paramore.Brighter/MessageMappers/JsonMessageMapper.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            IEvent => MessageType.MT_EVENT,
+            _ => throw new ArgumentException(@"This message mapper can only map Commands and Events", nameof(request))
+        };
+
+        if(publication.Topic is null)
+            throw new ArgumentException($"No Topic Defined for {publication}");
+
+        var header = new MessageHeader(messageId: request.Id, topic: publication.Topic, messageType: messageType);
+
+        var body = new MessageBody(JsonSerializer.Serialize(request, JsonSerialisationOptions.Options));
```
Comment (iancooper) [inline src/Paramore.Brighter/MessageMappers/JsonMessageMapper.cs:33]: In principle there is an async serializer. I am not sure if it actually has meaning. But we should investigate. The MS route here is to pass an enum flag to the common method and then just use sync or async code in the body, based on that flag as needed

Comment (iancooper) [reply src/Paramore.Brighter/MessageMappers/JsonMessageMapper.cs:33]: See the Flag Argument Hack here: https://learn.microsoft.com/en-us/archive/msdn-magazine/2015/july/async-programming-brownfield-async-development

**Review by iancooper — APPROVED** [review]
This looks really good. I can add a default Avro as well later

File: src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -379,7 +379,9 @@ public static MessageMapperRegistry MessageMapperRegistry(IServiceProvider provi
 
             var messageMapperRegistry = new MessageMapperRegistry(
                 new ServiceProviderMapperFactory(provider),
-                new ServiceProviderMapperFactoryAsync(provider)
+                new ServiceProviderMapperFactoryAsync(provider),
```
[the remark this answers is not in the document]
Comment (preardon) [reply src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionExtensions.cs:382]: @iancooper  what did we decide here?

File: src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionMessageMapperRegistry.cs (hunk 1 above)
Comment (preardon) [reply src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionMessageMapperRegistry.cs:44]: I think we need something set if we don't want to make this type nullable, I'm open either way

**Pull request merged**